internal/email/email.go

```diff
@@ -46,12 +63,42 @@ func NewMessage(
 		return nil, fmt.Errorf("error generating UUID: %w", err)
 	}
 
+	// Populate the template data source
+	data := bodyData{
+		AdminName:        sponsorDisplay,
+		OrganizationName: projectDisplay,
+		InvitationURL:    inviteURL,
+		RecipientEmail:   inviteeEmail,
+		MinderURL:        minderURLBase,
+		TermsURL:         DefaultMinderTermsURL,
+		PrivacyURL:       DefaultMinderPrivacyURL,
+		SignInURL:        minderURLBase,
+		RoleName:         role,
+		RoleVerb:         authz.AllRolesVerbs[authz.Role(role)],
+	}
+
+	// Validate the data source template for HTML injection attacks or empty fields
+	err = validateDataSourceTemplate(&data)
+	if err != nil {
+		zerolog.Ctx(ctx).Error().Err(err).Msg("error validating data source")
+		return nil, fmt.Errorf("error validating data source: %w", err)
```

Comment:
General rule of thumb for error handling: either pass the error up the chain, _or_ log the error and return `nil`.  If you do both, you'll get a lot of duplicate error messages in the logs, as each components logs the error _and_ passes it up to the next level for it to _also_ log the error.